### ISSUE TITLE
Unicode character is not displayed

### ISSUE THREAD

**Issue by stefanspringer1** [KaTeX/KaTeX#2226]
With the HTML page listed below (with KaTeX version 0.11.1 included from [http://cdn.jsdelivr.net](http://cdn.jsdelivr.net)), the following warning is displayed in the console for the character `&#x1E9E;`, but the character *is displayed*:

~~~Text
No character metrics for 'ẞ' in style 'Main-Regular' and mode 'text'
~~~

But for the character `&#x1D4B6;`, the following error message is thrown and the character is *not* displayed (the forumla and no further formula is rendered):

~~~Text
auto-render.min.js:1 Uncaught Error: Font metrics not found for font: .
~~~

From the documentation of the [Supported Functions](https://katex.org/docs/supported.html):

>If the KaTeX rendering mode is set to strict: false or strict: "warn" (default), then KaTeX will accept all Unicode letters in both text and math mode. 

Questions:

- With the setting `strict: false` in the HTML code below, there should not even be a warning for `&#x1E9E;`, is the configuration correct?
- Why then is `&#x1E9E;` rendered but not `&#x1D4B6;`?
- Even if the formula containing `&#x1E9E;` cannot be rendered, KaTeX should render subsequent formulas.
- *How then can I render a formula with any Unicode character?*

~~~HTML
<!DOCTYPE html>
<html lang="de">
<head>
  <meta charset="UTF-8"/>
  <link crossorigin="anonymous" href="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css" integrity="sha384-zB1R0rpPzHqg7Kpt0Aljp8JPLqbXI3bhnPWROx27a9N0Ll6ZP/+DiW/UqRcLbRjq" rel="stylesheet"></link>
  <script src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js" crossorigin="anonymous" defer="defer" integrity="sha384-y23I5Q6l+B6vatafAwxRu/0oK/79VlbSz7Q9aiSZUvyWYIYsd+qj+o24G5ZU2zJz"></script>
  <script src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js" crossorigin="anonymous" defer="defer" integrity="sha384-kWPLUVMOks5AQFrykwIup5lo0m3iMkkHrD0uJ4H5cjeGihAutqP0yW0J6dpFiVkI"></script>
  <script>
    document.addEventListener("DOMContentLoaded", function() {
      renderMathInElement(document.body, {
        strict: false
      });
    });
  </script>
</head>
<body>
  <p>\(a^2\)</p>
  <p>&#x1E9E;: \(&#x1E9E;\)</p>
  <p>&#x1D4B6;: \(&#x1D4B6;\)</p>
  <p>\(a^2\)</p>
</body>
</html>
~~~

**Comment by ronkok** [comment]
@stefanspringer1 You are correct. This should not result in an error message. KaTeX has a special handler for the Unicode code points in the [Mathematical Alphanumeric Symbols](https://www.unicode.org/charts/PDF/U1D400.pdf) range. It maps those code points to glyphs in the KaTeX fonts.

The KaTeX fonts do not include glyphs for lower case script letters. In that case, instead of an error, we should substitute a call to a system font. I'll work up a PR that fixes that.

You can avoid an error right now by calling `\mathscr{a}` or `\mathcal{a}`. I'm afraid you will be disappointed with that result, too. Lacking a KaTeX script glyph, KaTeX is apparently substituting an italic "*a*". It would be better if we called a script "𝒶" from a system font. I'm not sure I know how to fix that, but I'll look into it.

You might open an issue in the [KaTeX font repo](https://github.com/KaTeX/katex-fonts), asking for lower case script glyphs and calligraphic glyphs.

**Comment by stefanspringer1** [comment]
Thank you for your answer. I now asked for lowercase script letters in the KaTeX fonts in https://github.com/KaTeX/katex-fonts/issues/76. But as noted there, for such seldom used character I would say that referencing Unicode codepoints would suffice, but then the described problem should be resolved. I do not know enough to decide which way is better.

**Comment by edemaine** [comment]
The way to decide would be "what does LaTeX do?" Is there a LaTeX package that supports lower-case script letters?  Because default LaTeX renders `\mathscr{abc}` as italic letters, like KaTeX does currently.  To change that (odd) behavior, we'd want to match the behavior of LaTeX with some existing package.

**Comment by stefanspringer1** [comment]
> The way to decide would be "what does LaTeX do?" Is there a LaTeX package that supports lower-case script letters? Because default LaTeX renders `\mathscr{abc}` as italic letters, like KaTeX does currently. To change that (odd) behavior, we'd want to match the behavior of LaTeX with some existing package.

There seems to be an analogous issue with lowercase *calligraphic* letters. Concerning calligraphic letters, as noted on https://tex.stackexchange.com/questions/231322/how-to-get-the-lowercase-calligraphic-symbols, the packages bickham, boondox-cal, boondox-calo, and dutchcal have lowercase calligraphic letters which then can be accessed via \mathcal{a} etc. So in cases where people need calligraphic script letters, they would usually (after including an appropriate package) write them as \mathcal{a} etc. Using \mathscr{a} for a *real* lowercase script letter then does not feel "unnatural".

It is true that this would change the behavior of "pure" LaTeX but it seems that in those cases people tend to include appropriate packages to correct "odd" behaviours of standard LaTeX anyway.

**Comment by ronkok** [comment]
There is one more difficulty with script and calligraphic letters. Say that we use a system font to render the lower case letters with Unicode code points in the range from 1D4B6 to 1D4CF. In that scenario, we cannot be sure if the rendered glyph will be calligraphic or script.

Users with PCs, Macs, or iPhones will probably be served a glyph from the Cambria Math font. That font places calligraphic glyphs at those code points. Other mathematical fonts, like Asana or STIX, place script letters at those code points. You cannot be sure which one a reader will see.

I am even more unsure about Unix or Android. 

Two years ago, when we added support for Unicode code points, I often avoided the code points that had messy, ambiguous mappings to LaTeX. I'm inclined to avoid them here, as well.

Which means that the only clean way to support this request is by modifying the KaTeX fonts.
